### PR TITLE
Fix Network Service scenario, add unit tests

### PR DESCRIPTION
**Pull request by ehunter1878**

### CODE REVIEW

File: diagnosticsModule/Private/HelperUtilities.ps1 [hunk 1]
```diff
@@ -798,3 +798,14 @@ function IsExecutedByConnectHealth {
 
     return $true
 }
+
+function NullOrWhitespaceStringInArray($array)
+{
+    $EmptyElements = @($array | Where-Object { [string]::IsNullOrWhitespace($_) })
```
Comment (madhavpatel6) [inline diagnosticsModule/Private/HelperUtilities.ps1]: return (@($array | Where-Object { [string]::IsNullOrWhitespace($_) })).Count -ne 0

**Review by madhavpatel6 — CHANGES_REQUESTED** [review]
Small change

File: diagnosticsModule/Private/AdfsHealthChecks.ps1 [hunk 2]
```diff
@@ -390,19 +391,20 @@ Function TestADFSDuplicateSPN
                 return $testResult;
             }
         }
-        elseif ($adfsServiceAccount.Contains("LocalSystem"))
+        elseif ($adfsServiceAccount.Contains("LocalSystem") -or $adfsServiceAccount.Contains("NT AUTHORITY\NETWORK SERVICE"))
```
Comment (jodyhendrix) [inline diagnosticsModule/Private/AdfsHealthChecks.ps1]: We should make these checks case insensitive

File: diagnosticsModule/Test/Private/AdfsHealthChecks.Test.ps1 [hunk 3]
```diff
@@ -104,6 +104,161 @@ InModuleScope ADFSDiagnosticsModule {
         }
     }
 
+    Describe "TestADFSDuplicateSPN" {
+        
+        BeforeAll {
+            $_upnServiceAccount = "aadcsvc@contoso.com"
+            $_samServiceAccount = "contoso\aadcsvc"
+            $_localSystemAccount = "LocalSystem"
```
Comment (jodyhendrix) [inline diagnosticsModule/Test/Private/AdfsHealthChecks.Test.ps1:112]: Also add tests where the casing doesn't match.

Comment (ehunter1878) [reply diagnosticsModule/Test/Private/AdfsHealthChecks.Test.ps1:112]: Will do

File: diagnosticsModule/Private/AdfsHealthChecks.ps1 (hunk 2 above)
Comment (ehunter1878) [reply diagnosticsModule/Private/AdfsHealthChecks.ps1]: -eq is case insensitive.

**Review by madhavpatel6 — APPROVED** [review]

**Pull request merged**